### PR TITLE
Me/dpc 4166 remove hashed mbis

### PR DESCRIPTION
**Pull request by MEspositoE14s**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-4166

## 🛠 Changes

- Updated BBClient to so that patient searches by MBI are done with a post instead of a get, and the MBI is sent in the post body.
- Removed MBI hashing since it's no longer necessary.
- Removed BBClientV2 and its config.

## ℹ️ Context

The BFD team asked us to begin sending MBIs via post.  Instead of updating BBClientV2 along with BBClient, this PR removes it since it's not in use.

Ops changes: https://github.com/CMSgov/dpc-ops/pull/722

## 🧪 Validation

Ran locally and ensured we were able to pull data from BFD sandbox.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -456,7 +456,7 @@ If you want to run and debug integration tests through IntelliJ there are a few
   - You'll have one debugger tab open on an IT in dpc-api and another on the dependant service, allowing you to set break points in either and examine the test end to end.
 
 #### Running Integration Tests Against the BFD Sandbox
-Want to run your integration tests against the real BFD sandbox instead of using the MockBlueButtonClient?  In docker-compose.yml, under the aggregation service, set the USE_BFD_MOCK env variable to true and then rerun `make start-it-debug.`
+Want to run your integration tests against the real BFD sandbox instead of using the MockBlueButtonClient?  In docker-compose.yml, under the aggregation service, set the USE_BFD_MOCK env variable to false and then rerun `make start-it-debug.`
```
Comment (jdettmannnava) [inline README.md:459]: :sigh:

File: dpc-aggregation/src/test/java/gov/cms/dpc/aggregation/engine/AggregationEngineBFDClientTest.java [hunk 2]
```diff
@@ -68,14 +65,12 @@ public class AggregationEngineBFDClientTest {
     @BeforeEach
     public void setup() throws GeneralSecurityException {
         BlueButtonClient blueButtonClient = Mockito.spy(new BlueButtonClientImpl(bbClient, new BBClientConfiguration(), metricRegistry));
-        BlueButtonClientV2 blueButtonClientV2 = Mockito.spy(new BlueButtonClientV2Impl(bbClientV2, new BBClientConfiguration(), metricRegistry));
-        OperationsConfig config = new OperationsConfig(1000, tempDir.toString(), 1, 1, 1, YearMonth.now(), List.of(orgID.toString()));
+                OperationsConfig config = new OperationsConfig(1000, tempDir.toString(), 1, 1, 1, YearMonth.now(), List.of(orgID.toString()));
```
Comment (jdettmannnava) [inline dpc-aggregation/src/test/java/gov/cms/dpc/aggregation/engine/AggregationEngineBFDClientTest.java:68]: Odd formatting?

**Review by jdettmannnava — APPROVED** [review]
LGTM!
Can you deploy to dev and smoke test before merge?

File: dpc-aggregation/src/test/java/gov/cms/dpc/aggregation/engine/AggregationEngineBFDClientTest.java (hunk 2 above)
Comment (MEspositoE14s) [reply dpc-aggregation/src/test/java/gov/cms/dpc/aggregation/engine/AggregationEngineBFDClientTest.java]: Yeah, I'm not sure how that slipped past my linter.

**Pull request merged**